table/plain_table_reader.cc

```diff
@@ -114,13 +114,11 @@ PlainTableReader::~PlainTableReader() {
 
 Status PlainTableReader::Open(const ImmutableCFOptions& ioptions,
                               const EnvOptions& env_options,
+                              const PlainTableOptions& table_options,
                               const InternalKeyComparator& internal_comparator,
-                              unique_ptr<RandomAccessFile>&& file,
+                              std::unique_ptr<RandomAccessFile>&& file,
                               uint64_t file_size,
-                              unique_ptr<TableReader>* table_reader,
-                              const int bloom_bits_per_key,
-                              double hash_table_ratio, size_t index_sparseness,
-                              size_t huge_page_tlb_size, bool full_scan_mode) {
```

Comment:
This is a new feature that we needed for integrating this into our database.  Allows to pass a config string and get options instance back. I think this is the only one here.

From: Siying Dong [mailto:notifications@github.com]
Sent: Monday, July 6, 2015 3:45 PM
To: facebook/rocksdb
Cc: Dmitri Smirnov
Subject: Re: [rocksdb] Windows Port from Microsoft (#646)


In table/plain_table_reader.cc<https://github.com/facebook/rocksdb/pull/646#discussion_r33991164>:

>                                uint64_t file_size,

> -                              unique_ptr<TableReader>* table_reader,

> -                              const int bloom_bits_per_key,

> -                              double hash_table_ratio, size_t index_sparseness,

> -                              size_t huge_page_tlb_size, bool full_scan_mode) {

I like the change of passing through PlainTableOptions all the way down. But is it related to the port?

—
Reply to this email directly or view it on GitHub<https://github.com/facebook/rocksdb/pull/646/files#r33991164>.